os-client/src/main/kotlin/io/provenance/os/client/OsClient.kt

```diff
@@ -47,6 +57,33 @@ open class OsClient(uri: URI) {
 
         objectAsyncClient = ObjectServiceGrpc.newStub(channel)
         publicKeyBlockingClient = PublicKeyServiceGrpc.newBlockingStub(channel)
+        mailboxBlockingClient = MailboxServiceGrpc.newBlockingStub(channel)
+    }
+
+    fun ack(uuid: UUID) {
+        val request = Mailboxes.AckRequest.newBuilder()
+            .setUuid(Util.UUID.newBuilder().setValue(uuid.toString()).build())
+            .build()
+
+        mailboxBlockingClient.ack(request)
+    }
+
+    fun mailboxGet(publicKey: PublicKey, maxResults: Int): Collection<Pair<UUID, DIMEInputStream>> {
+        val mail = mutableListOf<Pair<UUID, DIMEInputStream>>()
+
+        val response = mailboxBlockingClient.get(
+            Mailboxes.GetRequest.newBuilder()
+                .setPublicKey(ECUtils.convertPublicKeyToBytes(publicKey).toByteString())
+                .setMaxResults(maxResults)
+                .build()
+        )
+
+        response.forEachRemaining {
```

Comment:
why forEachRemaining instead of just forEach... or better yet, why not just do `val mail = response.map {` or the like, instead of the mutable list?

Reply:
Yeah, I should be able to convert it to a sequence. Right now it's a java iterator so the methods exposed were odd.